numismatic/feeds/base.py

```diff
@@ -139,6 +146,85 @@ class RestClient(abc.ABC):
     cache_dir = attr.ib(default=None)
     requester = attr.ib(default='base')
 
+    def listen(self, symbol, channel, interval=1.0, exchange=None):
+        exchange = exchange if exchange else self.exchange
+        channel_name = f'{exchange}--{symbol}--{channel.__name__}'
+        logger.info(f'Subscribing to {channel_name} ...')
+        # timer
+
+        # FIXME: get the symbol properly
+        # application
+        def _get_raw_channel():
+            messages = channel(symbol[:3], symbol[3:], exchange=exchange,
+                               raw=True)
+            packet = '\n'.join(json.dumps(msg) for msg in messages)
+            return packet
+
+        channel_info = {'channel': channel.__name__}
+        subscription = Subscription(exchange=exchange,
+                                    symbol=symbol,
+                                    channel='ticker',
+                                    channel_info=channel_info, 
+                                    client=self,
+                                    listener=None,
+                                    handlers=self._get_handlers(),
+                                    )
+        # FIXME: find a better way to do this
+        subscription.listener = self._listener(subscription, interval=interval,
+                                               callback=_get_raw_channel)
+
+        # Install the raw_stream packet handler
+        subscription.raw_stream.sink(
+            partial(self.__handle_packet, subscription=subscription))
+        logger.info(f'Subscribed to {channel_name} ...')
+        return subscription
+             
+
+    @staticmethod
+    def __handle_packet(packet, subscription):
+        # most of the time we get json so only decode that once
+        try:
+            msg = json.loads(packet)
+        except:
+            msg = packet
+            raise
+        if not msg:
+            return
+        for handler in subscription.handlers:
+            result = handler(msg)
+            if isinstance(result, Event):
+                subscription.event_stream.emit(result)
+            elif result is STOP_HANDLERS:
+                break
+
+    @classmethod
+    def _get_handlers(cls):
+        return [getattr(cls, attr) for attr in dir(cls)
+                if callable(getattr(cls, attr)) 
+                and attr.startswith('parse_')]
```

Comment:
The node community uses the prefix `on`. So for example, `on_connect` and `on_trade`. It seems like it is a very short prefix and quite descriptive of the purpose. What do you think?

Reply:
So some of the python websocket client classes do this too. I see the point. If we go with the event handler publishing straight to the stream model then I think that sounds like a good convention, i.e. for the Websocket subscription handle_*() handlers. These ones just parse a message and return an event so slightly less appropriate here in my opinion but let me think about it for a bit and get back to you.